skyportal/handlers/api/instrument.py

```diff
@@ -17,16 +39,43 @@ def post(self):
             telescope_id, self.current_user, raise_if_none=True, mode="read"
         )
 
+        field_data = data.pop("field_data", None)
+        field_region = data.pop("field_region", None)
+
         schema = Instrument.__schema__()
         try:
             instrument = schema.load(data)
         except ValidationError as exc:
             return self.error(
                 'Invalid/missing parameters: ' f'{exc.normalized_messages()}'
             )
-        instrument.telescope = telescope
-        DBSession().add(instrument)
-        self.verify_and_commit()
+
+        try:
```

Comment:
This is fine, but we typically just use `query.first()` and then check if the value is `None` (simpler than try/except)

Reply:
Fixed I hope.

Reply:
Lolz I didn't understand what this meant I think. Maybe you can suggest a change?